### PR TITLE
Deny direct command execution on privileged containers

### PR DESCRIPTION
**Pull request by ytsarev**
Deny host access through privileged containers with the help of
https://kubernetes.io/docs/admin/admission-controllers/#denyescalatingexec

### CODE REVIEW

**Comment by mumoshu** [comment]
@ytsarev Thanks for the PR 👍 
I basically like this change except one concern about debugging.
I have been using `kubectl exec/attach` to debug privileged pods - how can we do that after this change? Logging-in to the node via `ssh` and then running  e.g. `docker exec`?

**Comment by danielfm** [comment]
This is a nice feature, but I think it should be configurable by the user (and disabled by default).

**Comment by ytsarev** [comment]
@mumoshu yes, debugging privileged would be quite inconvenient as you described.
So I will make it as a config option as @danielfm suggested

**Comment by mumoshu** [comment]
@ytsarev @danielfm Thanks! Yes, I agree with you. I'd suggest adding `experimental.admission. denyEscalatingExec.enabled` to `cluster.yaml` for now.

**Comment by ytsarev** [comment]
@mumoshu configuration option added, please review.
Also gave it an e2e test, the option works as expected
```
 kubectl exec -it kube-system-kube2iam-kube2iam-5gk4n -n kube-system -- sh
Error from server (InternalError): Internal error occurred: [cannot exec into or attach to a privileged container, object does not implement the Object interfaces]
```

File: core/controlplane/config/templates/cloud-config-controller [hunk 1]
```diff
@@ -1468,7 +1468,7 @@ write_files:
           - --authentication-token-webhook-cache-ttl={{ .Experimental.Authentication.Webhook.CacheTTL }}
           {{ end }}
           - --advertise-address=$private_ipv4
-          - --admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass{{if .Experimental.Admission.PodSecurityPolicy.Enabled}},PodSecurityPolicy{{ end }},ResourceQuota
+          - --admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass{{if .Experimental.Admission.PodSecurityPolicy.Enabled}},PodSecurityPolicy{{ end }},ResourceQuota,{{if .Experimental.Admission.DenyEscalatingExec.Enabled}}DenyEscalatingExec{{end}}
```
Comment (mumoshu) [inline core/controlplane/config/templates/cloud-config-controller]: I guess that the comma should go inside the block like `{{if .Experimental.Admission.DenyEscalatingExec.Enabled}},DenyEscalatingExec{{end}}`.
Anyway, I'll fix it myself myif my assumption turned to be correct 😉

**Comment by mumoshu** [comment]
LGTM. @ytsarev Thank you very much for your contribution 👍

File: core/controlplane/config/templates/cloud-config-controller (hunk 1 above)
Comment (mumoshu) [inline core/controlplane/config/templates/cloud-config-controller]: JFYI: The trailing comma was ok. My apiserver is running without any problem with:
```
--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,ResourceQuota,
```

Comment (ytsarev) [inline core/controlplane/config/templates/cloud-config-controller]: Good to know, thanks

**Pull request merged**